Tactical/Soldier Control.cpp

```diff
@@ -17775,44 +17775,47 @@ bool SOLDIERTYPE::AddBestFlashLight()
 
 	struct position_2d
 	{
+        INT16 x, y;
+
 		position_2d(INT32 gridNo)
 		{
 			ConvertGridNoToXY(gridNo, &x, &y);
 		}
-		position_2d(INT16 _x, INT16 _y)
-		{
-			x = _x;
-			y = _y;
-		}
-		INT16 x, y;
+		position_2d(INT16 _x, INT16 _y) : x{_x}, y{_y}
+        {
+        }
 	};
 	struct vector_2d
 	{
+        INT16 dx, dy;
+        float length;
+
 		vector_2d(INT8 direction)
 		{
 			ConvertDirectionToVectorInXY(direction, &dx, &dy);
-			length = sqrt(pow(dx, 2) + pow(dy, 2));
+            length = CalcLength(dx, dy);
 		}
 		vector_2d(position_2d from, position_2d to)
 		{
 			dx = to.x - from.x;
 			dy = to.y - from.y;
-			length = sqrt(pow(dx, 2) + pow(dy, 2));
+            length = CalcLength(dx, dy);
 		}

```

Comment:
missed this one:

```cpp
vector_2d(position_2d from, position_2d to) : dx{to.x - from.x}, dy{to.y - from.y}, length{CalcLength(dx, dy)} {}
```

Reply:
I dont want to do calculations as part of that initializer syntax to avoid having issues when breakpointing